### DISCUSSION TITLE
How to create workflow from yaml?

### DISCUSSION THREAD

**Discussion by rainfd** [argoproj-labs/hera#781] [Q&A] [answered]
I want to create workflow from a exist yaml file.
Here is the code
```go
  wf = Workflow.from_yaml(template)
  req = WorkflowCreateRequest(workflow=wf)
  client.create_workflow(req, namespace=namespace)
```
WorkflowCreateRequest(workflow=wf) wf type mismatch, wf is type ModelMapperMixin, not type Workflow.
Is there any example to create workflow in correct way?

**Comment by elliotgunton** [comment, marked as answer]
You can call `wf.create()` directly if you have set up a `workflows_service` on the `Workflow`. You shouldn't need to use `WorkflowCreateRequest` which is an auto-generated model class with limitations.

We can create the workflow in the same way as the https://hera.readthedocs.io/en/stable/walk-through/quick-start/#hello-world example - you can manipulate the `Workflow` object after loading it from yaml:

```py
w = Workflow.from_yaml(template)
w.namespace="argo"  # if not in the yaml
w.workflows_service=WorkflowsService(host="https://localhost:2746")  # this property is Hera-specific
w.create()
```
